AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1249,6 +1249,9 @@ - (void)addSubnode:(ASDisplayNode *)subnode
   // If we are a managed hierarchy, as in ASCellNode trees, it will also apply our .interfaceState.
   [subnode __setSupernode:self];
 
+  // propagate our traits to the child we are about to add.
+  ASEnvironmentStatePropagateDown(subnode, [self environmentTraitCollection]);
```

Comment:
would it make more sense to do this in _setSupernode: ?  If we add additional call points, this could be left out accidentally.  It would seem to be an action that should take place whenever a parent is added.